include/common/internal/log.h

```diff
@@ -0,0 +1,30 @@
+#ifndef LOG_H__
+#define LOG_H__
+
+#include <thread>
+#include <chrono>
+#include <sstream>
+#include <mutex>
+#include <vector>
+#include <iomanip>
+
+typedef void(*logCallbackFunction) (const char *message);
+
+extern std::mutex logMutex;
+
+#define DEBUG(message) do { \
```

Comment:
Please avoid using `DEBUG` macro in a header file, other projects might be using the same (like mine :). Even better if we could avoid macros altogether unless strictly necessary (which I don't think it is in this case)

Reply:
I do agree, instead of `DEBUG` a pc-ble-driver specific name should be used. And that is also true that macro in this case doesn't seem to make much sense.

What macro in this case provides is to make it easy to pass any chain that can be inserted to a stream. However the original solution I suggested was very different. There is no alternative to macros if we want to have location info (e.g. `__FILE__`, `__LINE__`, or similar).

Also note that `DEBUG` inserts everything to a stream, then take the resulting string out which will be inserted to an other stream in `log()` function. Some cases this is also done outside of `DEBUG`. Please make is simpler, there is no need for this much complexity and overhead. My suggestion is to remove the `log()` function, and remove the `std::stringstream`s and `setLogCallback()`, wrap what's necessary in something like `PC_BLE_DEBUG`.

Reply:
@torfinnberset Thank you for reviewing. You have a point, the DEBUG macro with the small log api was lying around and put directly into the project just to be used in the integration tests. I've made a new commit where a NRF_LOG macro is defined instead so it will not crash with compiler or users DEBUG defines.

I'm a bit curious, do you need to include pc-ble-driver internal header files into your project?

@bencefr Well, I received it and used it. But I think the header file log solution I committed now is more aligned with what you were thinking.